tutorial/source/gmm.ipynb

```diff
@@ -248,71 +250,189 @@
VLQiRUrOGc04gg0GzFCdY6wcjq9np+aN8ft99+pcvGi6hwh/pMsWJwImQzDtsTHm8mcuSxG4zl8fMayefNXqpNEMuTJ05y7d/+gaNEOXL68QHWOECIFDapShQmHD9Myb15W3LqlOkekB2fPQpkyYGcHt26BhyxsL1KfkgWLhUgPvv12LUbjOcCVuXN7qM4RyeTnNwQoSFBQda5dU10jhEgpJqORpUePAtC2XTvFNSLdKF0aqlWD+Hi0X39VXSPEv5KBlrBpZrPGTz/9AEDNmr3Il0/2UKY3HTpUoW7dK2jaF0yQU7WEsBnbxo7lttlMDp2OhrJIsUiGuy1a0A2oMXw4mtmsOkeIV5KBlrBpEyYEEB19GMjE3Ll9VeeI1/TttwbAMj1/cLDiGCFEipg3ezYA7b28cJTD9EUyOLduzRJg/5MnHJw/X3WOEK8kAy1h08aNmwJAhQpdKVUql+Ia8bpq14Zq1eKIjV1Eu3bTVOcIId5USAhVg4MpBHQZNkx1jUhnXAsUoGWRIgD8Ioc6CCsmAy1hs/bvhwcPFqHXf8esWQNV54g3oNOBj892oAPbt3/L1asPVScJId7EokV8ZTZzpWpVyjZrprpGpEOde/cGYOn580TLmlrCSslAS9gsPz+A7HToMJS33y6gOke8oeHDfXByKgc8olu36apzhBCvS9MsxwED+m7dFMeI9KpWr14UsbPjEbBq6FDVOUIkSgZawibt2/eIdes0dDr4SmZztwl6vY6uXS0nzO/YMZng4MeKi4QQr+PkwoWsvHgRo7MztG6tOkekU3o7OzrVrg3AL8uXK64RInEy0BI2qWXLDkB16tU7SsmSqmtEShk/vhX29sXQtAd06iR7tYRIjyZ+/z2tAN+CBSFrVtU5Ih3rMGYMOmBXRARXAgJU5wjxEhloCZuzfPlx7t79AzhIz55OqnNECnJysqNjR8uJ81u3juPOnUeKi4QQyRFx4wa/X7kCQNt+/RTXiPQuf7VqdMmblxFAlnXrVOcI8RIZaAmb07//KAAKFGhNkyalFdeIlDZ16mfY25dA08Lo2FFmIBQiPVn69dc8AUo7OvJ2166qc4QNmDtpEiMBj99/B6NRdY4QL5CBlrApS5ce4+7dNYCO6dOHq84RqcDJyY4uXYYDtdm/vxaRkaqLhBBJoZnNzF2zBoAuDRqg08uvICIFNG0KuXNbFllctUp1jRAvkFc5YVN8fUcCULBgGz76qJTaGJFqpk37jJIld/Ho0TtMnaq6RgiRFIFz53L0yRMcgXZjx6rOEbbC3h5T9+6sBfr5+qquEeIFVjHQmj59OoUKFcLJyYlq1apx8ODBf91+xYoVvPXWWzg5OVGuXDk2bdr0wtc7duyITqd74eLj45OaT0FYgSVLjhIcvA7QM2OGLIBpy+zsdAx/usPS3x8iItT2CCH+25QxYwBoW7w4uUrJG2Ei5dxr3pxWwJTgYI4sXqw6R4gEygday5cvx9fXlxEjRnD06FEqVKhAgwYNCA0NTXT7ffv20aZNG7p06cKxY8do2rQpTZs25fTp0y9s5+Pjw927dxMuS5cuTYunIxQaOvQXAAoVakOjRm8prhGprXVrKF78PuHhQ/jkkwmqc4QQ/8J07Rr3bt4EoO/TAZcQKcWzfHlaFSwIwLSRI9XGCPEc5QOtiRMn0q1bNzp16kTp0qWZNWsWzs7O/PLLL4luP2XKFHx8fBg0aBClSpXiu+++o1KlSvz0008vbOfo6Iinp2fCJXv27GnxdIQigYFw7doUdLplzJo1QnWOSAMGAzRsGACMZdu277lyJUx1khDiFQyzZ7MDOF+lCuVbtVKdI2xQ76eLFi+7coV7584prhHCQulAy2g0cuTIEerWrZtwm16vp27dugQGBiZ6n8DAwBe2B2jQoMFL2+/atQt3d3dKlixJjx49ePDgwSs7YmNjiYyMfOEi0g9Ne7YosYHOnVvToEFx1Ukijfj7t8LJqRwQwaef+qnOEUIk5skTmDMHgJJDhiiOEbaqWufOVHZ2JhaYJ0sHCCuhdKB1//59TCYTHh4eL9zu4eFBcHBwovcJDg7+z+19fHxYtGgRAQEB/Pjjj+zevZuGDRtiMpkSfUw/Pz9cXV0TLvnz53/DZybS0pw5F/n77yicnECOGMhY7Oz0fP215aT6w4enERh4Q3GREOKfjvv5cT8sDAoVgiZNVOcIG6XT6+ndti0AMwMCiI+JUVwkhBUcOpgaPv30U5o0aUK5cuVo2rQpGzZs4NChQ+zatSvR7YcMGUJERETC5ebT48iF9TMaTfTt2xwoTsuW+8mXT3WRSGvDhjUkW7b3gFjatZMp/YWwJvTtvQAAIABJREFUJprZTLtx48gPbK1Xz3LMrxCp5JPx48ml03HTZGLtMJkUS6indKDl5uaGwWAgJCTkhdtDQkLw9PRM9D6enp7J2h6gSJEiuLm5cfny5US/7ujoiIuLywsXkT58+eWvxMaeQaeLYfTokqpzhAJ6vY5Jk8YBcOXKIlauPKm4SAjxzM5JkzgdG4sBqPbNN6pzhI1zcnWlW/XqVARybN5sObdACIWUDrQcHBzw9vYmICAg4Taz2UxAQADVq1dP9D7Vq1d/YXuAP//885XbA9y6dYsHDx6QO3fulAkXViEs7AkLFljesWrU6BsKF5YJTzKqjh2rkC/fJ4BGz56yV0sIazF5/HgAOpQpQ7ZChdTGiAxh2OLFHLG35/0zZ+Dvv1XniAxO+aGDvr6+zJ07l4ULF3Lu3Dl69OhBVFQUnTp1AqB9+/YMee7k2b59+7Jlyxb8/f05f/48I0eO5PDhw/Tq1QuAx48fM2jQIPbv38+1a9cICAjg448/plixYjRo0EDJcxSpo3376ZhMtzAY8rN4cS/VOUKx+fPHoNN1JTT0J3buVF0jhDi5ciXrQ0LQAb3HjVOdIzIIp8KF0XXpYrnyww9qY0SGp3yg1bp1ayZMmMDw4cPx8vLi+PHjbNmyJWHCixs3bnD37t2E7
```

Comment:
You can use a progress bar here (if you prefer), by:

```python
from tqdm import trange
...
for i in trange(300 if not smoke_test else 2):
    ...
```
You can use a better looking progress bar by importing `tqdm` from `tqdm.autonotebook`, but that would also throw an experimental warning which you might want to filter.